[2.2.0 regression] mysqlclient-sys does not provide bundled bindings for libmysqlclient 21. Consider using the `buildtime_bindgen` feature or contribute bindings to the crate

## Setup

Regular build.

### Versions

- **Rust:** rust-1.78.0
- **Diesel:** 2.2.0
- **Operating System** FreeBSD 14.0

## Problem Description

```
[mysqlclient-sys 0.3.1] thread 'main' panicked at /wrkdirs/usr/ports/databases/diesel/work/diesel-2.2.0/cargo-crates/mysqlclient-sys-0.3.1/build.rs:160:9:
[mysqlclient-sys 0.3.1] mysqlclient-sys does not provide bundled bindings for libmysqlclient 21. Consider using the `buildtime_bindgen` feature or contribute bindings to the crate
[mysqlclient-sys 0.3.1] stack backtrace:
[mysqlclient-sys 0.3.1]    0: rust_begin_unwind
[mysqlclient-sys 0.3.1]    1: core::panicking::panic_fmt
[mysqlclient-sys 0.3.1]    2: build_script_build::parse_version
[mysqlclient-sys 0.3.1]    3: build_script_build::main
[mysqlclient-sys 0.3.1]    4: core::ops::function::FnOnce::call_once
[mysqlclient-sys 0.3.1] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: failed to run custom build command for `mysqlclient-sys v0.3.1`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `CARGO=/usr/local/bin/cargo CARGO_CFG_PANIC=unwind CARGO_CFG_TARGET_ABI='' CARGO_CFG_TARGET_ARCH=x86_64 CARGO_CFG_TARGET_ENDIAN=little CARGO_CFG_TARGET_ENV='' CARGO_CFG_TARGET_FAMILY=unix CARGO_CFG_TARGET_FEATURE=fxsr,sse,sse2 CARGO_CFG_TARGET_HAS_ATOMIC=16,32,64,8,ptr CARGO_CFG_TARGET_OS=freebsd CARGO_CFG_TARGET_POINTER_WIDTH=64 CARGO_CFG_TARGET_VENDOR=unknown CARGO_CFG_UNIX='' CARGO_ENCODED_RUSTFLAGS='-Clink-arg=-L/usr/local/lib-Clink-arg=-fstack-protector-strong' CARGO_FEATURE_DEFAULT=1 CARGO_MANIFEST_DIR=/wrkdirs/usr/ports/databases/diesel/work/diesel-2.2.0/cargo-crates/mysqlclient-sys-0.3.1 CARGO_MANIFEST_LINKS=mysqlclient CARGO_PKG_AUTHORS='' CARGO_PKG_DESCRIPTION='Auto-generated rust bindings for libmysqlclient' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=mysqlclient-sys CARGO_PKG_README=README.md CARGO_PKG_REPOSITORY='https://github.com/sgrif/mysqlclient-sys' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.3.1 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=3 CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_PRE='' DEBUG=false HOST=x86_64-unknown-freebsd LD_LIBRARY_PATH='/wrkdirs/usr/ports/databases/diesel/work/target/release/deps:/wrkdirs/usr/ports/databases/diesel/work/target/release:/usr/local/lib/rustlib/x86_64-unknown-freebsd/lib' NUM_JOBS=6 OPT_LEVEL=2 OUT_DIR=/wrkdirs/usr/ports/databases/diesel/work/target/x86_64-unknown-freebsd/release/build/mysqlclient-sys-61f26f1caf885244/out PROFILE=release RUSTC=/usr/local/bin/rustc RUSTC_LINKER=cc RUSTDOC=/usr/local/bin/rustdoc TARGET=x86_64-unknown-freebsd /wrkdirs/usr/ports/databases/diesel/work/target/release/build/mysqlclient-sys-6bcf5f36c20e6c65/build-script-build` (exit status: 101)
  --- stdout
```


### What are you trying to accomplish?


### What is the expected output?


### What is the actual output?


### Are you seeing any additional errors?


### Steps to reproduce

<!--
Please include as much of your codebase as needed to reproduce the error.  If the relevant files are large, please consider linking to a public repository or a [Gist](https://gist.github.com/). This includes normally the following parts:

* The exact code where your hit the problem
* Relevant parts your schema, so any `table!` macro calls required for
* Any other type definitions involved in the code, which produces your problem
-->

while it works for macos CI jobs, but it does not work linux CI, any idea?

```
    Unable to generate bindings: ClangDiagnostic("bindings/wrapper.h:4:10: fatal error: 'mysql/mysql.h' file not found\n")
```

linux ci run log, https://github.com/Homebrew/homebrew-core/actions/runs/9339158533/job/25703533805?pr=173312

--- a/diesel/Cargo.toml
+++ b/diesel/Cargo.toml
@@ -25,7 +25,7 @@ byteorder = { version = "1.0", optional = true }
 chrono = { version = "0.4.20", optional = true, default-features = false, features = ["clock", "std"] }
 libc = { version = "0.2.0", optional = true }
 libsqlite3-sys = { version = ">=0.17.2, <0.29.0", optional = true, features = ["bundled_bindings"] }
-mysqlclient-sys = { version = ">=0.2.5, <0.4.0",  optional = true }
+mysqlclient-sys = { version = ">=0.2.5, <0.5.0",  optional = true }
 mysqlclient-src = { version = "0.1.0", optional = true }
 pq-sys = { version = ">=0.4.0, <0.7.0", optional = true }
 pq-src = { version = "0.3", optional = true }

--- a/diesel_cli/Cargo.toml
+++ b/diesel_cli/Cargo.toml
@@ -47,7 +47,7 @@ url = { version = "2.2.2" }
 libsqlite3-sys = { version = ">=0.17.2, <0.29.0", optional = true }
 pq-sys = { version = ">=0.4, <0.7.0", optional = true }
 openssl-sys = { version = "0.9.93", features = ["vendored"], optional = true }
-mysqlclient-sys = { version = ">=0.2.5, <0.4.0", optional = true }
+mysqlclient-sys = { version = "0.4.0", optional = true }
 diffy = "0.3.0"
 regex = "1.0.6"
 serde_regex = "1.1"

--- a/diesel_tests/Cargo.toml
+++ b/diesel_tests/Cargo.toml
@@ -24,7 +24,7 @@ rand = "0.8.4"
 libsqlite3-sys = { version = "0.28", optional = true }
 pq-sys = { version = "0.6", optional = true }
 pq-src = { version = "0.3", optional = true }
-mysqlclient-sys = { version = "0.3", optional = true }
+mysqlclient-sys = { version = "0.4", optional = true }
 mysqlclient-src = { version = "0.1.0", optional = true }
 
 [features]

